Singularity/Singularity/Manager/DistributionManager.cs

```diff
@@ -1151,6 +1151,17 @@ public void Unregister(List<PlatformBlank> platforms, bool isDef, bool inactivat
             mHandler?.ForceSliderPages();
 
         }
+
+        /// <summary>
+        /// This will be called from the SliderHandler when it changes.
+        /// It just unregisters its reference, so the DistributionManager can't communicate with it anymore.
+        /// </summary>
+        /// <param name="handler"></param>
+        internal void Unregister(SliderHandler handler)
+        {
+            mHandler = null;
```

Comment:
so why do you take a handler as an argument? Did you mabye intend to check for equality and if they are unreference it? Every handler could right now theoretically Unregister the currently registered one. (Security-wise important)

Reply:
because the handler wants to unregister itself when there is a change, because the handler will register at a new DistrManager and therefore should not be registered at an old one..

Reply:
yes, but handler and mHandler can also be totally unrelated, right? everyone could call `Unregister(null)` and whatever handler would be registered at this point would be unregistered. Is it accounted for that mHandler can be null? This would be a perfect opportunity for an Optional<SliderHandler> :)

Reply:
Ooh, I get the point.. Yes, absolutely.. I'll fix it..